File: articles/grace-hopper-compilers-and-the-humanization-of-programming/haskell/CompilerPipeline.hs
module Main where

main :: IO ()
main = do
  let source = "ADD PAYROLL-TOTAL TO TAX-BASE"
  let tokens = words source
  putStrLn ("source=" ++ source)
  putStrLn ("tokens=" ++ show tokens)
  putStrLn "target_code=machine-specific instruction sequence"

File: articles/grace-hopper-compilers-and-the-humanization-of-programming/go/compiler_pipeline.go
package main

import (
	"fmt"
	"strings"
)

func main() {
	source := "ADD PAYROLL-TOTAL TO TAX-BASE"
	tokens := strings.Fields(source)
	fmt.Println("source=" + source)
	fmt.Printf("tokens=%v\n", tokens)
	fmt.Println("target_code=machine-specific instruction sequence")
}
